lib/system.rb

```diff
@@ -151,7 +151,15 @@ def locale
     @locale || "C"
   end
 
-  def rpm_database
-    @rpm_database ||= RpmDatabase.new(self)
+  def managed_files_database
+    if @managed_files_database
+      return @managed_files_database
+    else
+      if has_command?("rpm")
+        @managed_files_database = RpmDatabase.new(self)
+      else
```

Comment:
is it safe to assume everything else falls into the dpkg group? Maybe we could specifically have the cases for rpm and dpkg and anything else would raise